types/queries.d.ts

```diff
@@ -12,14 +14,14 @@ export type AllByBoundAttribute<T extends HTMLElement = HTMLElement> = (
   container: HTMLElement,
   id: Matcher,
   options?: MatcherOptions,
-) => T[]
+) => NonEmptyArray<T>
```

Comment:
I think we can't change this because `queryAllBy` queries return an empty array when there are no matches.
https://codesandbox.io/s/react-testing-library-demo-forked-o2sw6?file=/src/__tests__/hello.js